var/spack/repos/builtin/packages/r-rmariadb/package.py

```diff
@@ -0,0 +1,29 @@
+# Copyright 2013-2020 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+from spack import *
+
+
+class RRmariadb(RPackage):
+    """Implements a 'DBI'-compliant interface to 'MariaDB'
+    (<https://mariadb.org/>) and 'MySQL' (<https://www.mysql.com/>)
+    databases."""
+
+    homepage = "https://rmariadb.r-dbi.org/"
+    url      = "https://cloud.r-project.org/src/contrib/RMariaDB_1.0.8.tar.gz"
+    list_url = "https://cran.r-project.org/src/contrib/Archive/RMariaDB"
+
+    version('1.0.8', sha256='3c8aedc519dc063ceb068535a3700bc5caf26f867078cc5a228aa8961e2d99f5')
+
+    depends_on('r@2.8.0:', type=('build', 'run'))
+    depends_on('r-bit64', type=('build', 'run'))
+    depends_on('r-dbi@1.1.0:', type=('build', 'run'))
+    depends_on('r-hms@0.5.0:', type=('build', 'run'))
+    depends_on('r-rcpp@0.12.4:', type=('build', 'run'))
+    depends_on('r-bh', type=('build', 'run'))
+    depends_on('r-plogr', type=('build', 'run'))
+
+    # non-R dependencies
+    depends_on('mariadb-c-client')
```

Comment:
I would add a note saying it can also be built against mysql. I'm not sure if we should handle that through virtual dependencies or variants, but a comment is fine for now.

Reply:
Good point. It could also use the mariadb package so there are a few options. I think the version constraints get complicated for those. I did not realize that there was a mysql-client provider. That would seem to be the best option. I do not see mariadb in the list though.
```console
$ spack providers mysql-client
mysql-client:
mariadb-c-client  mysql
```
I will play around with that a bit so maybe mark this PR WIP for now.

Reply:
Oh wait, there is a mariadb-client virtual package as well.
```console
mariadb-client:
mariadb  mariadb-c-client
```
Maybe the mariadb-c-client package should not be in the mysql-client provider list?

Reply:
I'm not too familiar with it, but aren't mysql and mariadb more or less interchangeable APIs? I think they should be unified into a single virtual provider.

Reply:
The code bases diverged and I know that I have had issues with mariadb-5 vs mariadb-10, which prompted my earlier comment about version constraints. I am guessing that there is a reason why spack has both mysql-client and mariadb-client virtual packages. I will poke at it a bit to try to understand it more but it might be a few days before I can get to it.

Reply:
I checked and verified that any of mysql, mariadb, or mariadb-c-client can be used for this. I changed the dependency to `depends_on('mariadb-client')`. I think that it would be worthwhile to determine whether the `mysql-client` and `mariadb-client` virtual packages could/should be consolidated, but that seems like another PR.